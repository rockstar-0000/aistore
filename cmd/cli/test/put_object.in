echo "0123456789" > $OBJECT_1.txt
ais bucket create ais://$BUCKET_1 | awk '{print $1 " " $2}'

ais put $OBJECT_1.txt ais://$BUCKET_2/object_1.txt // FAIL "bucket "ais://$BUCKET_2" does not exist"
ais put $OBJECT_1.txt cloud://$BUCKET_1/object_1.txt // FAIL "invalid backend provider "cloud""

head -c 31971520 /dev/urandom | ais object put - ais://$BUCKET_1/rand.txt
ais put $OBJECT_1.txt ais://$BUCKET_1/object_1.txt
ais put $OBJECT_1.txt ais://$BUCKET_1/cksum --crc32c 0767345f
ais put $OBJECT_1.txt ais://$BUCKET_1/cksum --crc32c badchecksum // FAIL "BAD DATA CHECKSUM: crc32c(badchecksum != 0767345f)"
ais object put $OBJECT_1.txt ais://$BUCKET_1/cksum --md5 3749f52bb326ae96782b42dc0a97b4c1
ais object put $OBJECT_1.txt ais://$BUCKET_1/cksum --md5 badchecksum // FAIL "BAD DATA CHECKSUM: md5(badchecksum != 3749f52bb326ae96782b42dc0a97b4c1)"
ais object put $OBJECT_1.txt ais://$BUCKET_1/cksum --sha256 c67c199595622dfbdc9e415c4a0ad6166eb49cbf74c6aac7bb3e958604d5ecb8
ais object put $OBJECT_1.txt ais://$BUCKET_1/cksum --sha256 badchecksum // FAIL "BAD DATA CHECKSUM: sha256(badchecksum != c67c199595622dfbdc9e415c4a0ad6166eb49cbf74c6aac7bb3e958604d5ecb8)"
ais object put $OBJECT_1.txt ais://$BUCKET_1/cksum --sha512 e7da5269d4cd882deb8d7b7ca5cbf424047f56815fd7723123482e2931823a68d866627a449a55ca3a18f9c9ba7c8bb6219a028ba3ff5a5e905240907d087e40
ais object put $OBJECT_1.txt ais://$BUCKET_1/cksum --sha512 badchecksum // FAIL "BAD DATA CHECKSUM: sha512(badchecksum != e7da5269d4cd882deb8d7b7ca5cbf424047f56815fd7723123482e2931823a68d866627a449a55ca3a18f9c9ba7c8bb6219a028ba3ff5a5e905240907d087e40)"
ais object put $OBJECT_1.txt ais://$BUCKET_1/cksum --compute-checksum
ais put $OBJECT_1.txt ais://$BUCKET_1/cksum --xxhash 05967d5390ac53b0
ais put $OBJECT_1.txt ais://$BUCKET_1/cksum --xxhash badchecksum // FAIL "BAD DATA CHECKSUM: xxhash(badchecksum != 05967d5390ac53b0)"

ais object put $OBJECT_1.txt ais://$BUCKET_1/object_2.txt
ais bucket ls ais://$BUCKET_1 --props size,version

echo "0" > $OBJECT_1.txt
mkdir -p /tmp/dir1/dir2 && echo "0123456789" > /tmp/dir1/dir2/object_3.txt

ais put $OBJECT_1.txt ais://$BUCKET_1/object_1.txt // Bump the version of 'object_1.txt' object
ais put /tmp/dir1/dir2/object_3.txt ais://$BUCKET_1
ais bucket ls ais://$BUCKET_1 --props size,version

rm -rf /tmp/dir1

# setup files structure for next commands
for d1 in {0..4}; do for d2 in {0..4}; do mkdir -p /tmp/rdir/test$d1/$d2 && echo -n "0" > /tmp/rdir/test$d1/$d2/test.txt; done; done

# test range absolute paths

ais bucket create ais://$BUCKET_2 | awk '{print $1 " " $2}'
ais object put "/tmp/rdir/test{0..4}/{0..4}/test.txt" ais://$BUCKET_2 -y // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais object get ais://$BUCKET_2/test$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

ais object put "/tmp/rdir/test{0..4}/{0..4}/test.txt" ais://$BUCKET_2/subdir/ -y // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais object get ais://$BUCKET_2/subdir/test$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

# test range relative paths

ais bucket create ais://$BUCKET_3 | awk '{print $1 " " $2}'
cd /tmp && ais object put "rdir/test{0..2}/{0..2}/test.txt" ais://$BUCKET_3 -y // IGNORE
RES=0 && for d1 in {0..2}; do for d2 in {0..2}; do ais object get ais://$BUCKET_3/test$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES
cd /tmp && ais object put "./rdir/test{0..2}/{0..2}/test.txt" ais://$BUCKET_3/subdir -y // IGNORE
RES=0 && for d1 in {0..2}; do for d2 in {0..2}; do ais object get ais://$BUCKET_3/subdirtest$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

# test put * absolute paths

ais bucket create ais://$BUCKET_4 | awk '{print $1 " " $2}'
ais object put "/tmp/rdir/*.txt" ais://$BUCKET_4 -y -r // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais object get ais://$BUCKET_4/test$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES
ais object put "/tmp/rdir/*.txt" ais://$BUCKET_4/subdir/ -y -r // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais object get ais://$BUCKET_4/subdir/test$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

# test put * relative paths

ais bucket create ais://$BUCKET_5 | awk '{print $1 " " $2}'
cd /tmp && ais object put "rdir/*.txt" ais://$BUCKET_5 -y -r // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais object get ais://$BUCKET_5/test$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES
cd /tmp && ais object put "rdir/*.txt" ais://$BUCKET_5/subdir -y -r // IGNORE
RES=0 && for d1 in {0..4}; do for d2 in {0..4}; do ais object get ais://$BUCKET_5/subdirtest$d1/$d2/test.txt /dev/null && RES=$((RES+1)); done; done && echo $RES

rm -rf /tmp/rdir // IGNORE

ais bucket rm --yes --ignore-error ais://$BUCKET_1 ais://$BUCKET_2 ais://$BUCKET_3 ais://$BUCKET_4 ais://$BUCKET_5 // IGNORE
