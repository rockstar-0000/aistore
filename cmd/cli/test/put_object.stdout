"ais://$BUCKET_1" created
^PUT.*=> ais://$BUCKET_1/rand.txt$

^PUT.*=> ais://$BUCKET_1/object_1.txt$

^PUT.*=> ais://$BUCKET_1/cksum$

^PUT.*=> ais://$BUCKET_1/cksum$

^PUT.*=> ais://$BUCKET_1/cksum$

^PUT.*=> ais://$BUCKET_1/cksum$

^PUT.*=> ais://$BUCKET_1/cksum$

^PUT.*=> ais://$BUCKET_1/cksum$

^PUT.*=> ais://$BUCKET_1/object_2.txt$

NAME			 SIZE	 VERSION
cksum		 11B	 4
object_1.txt		 11B	 1
object_2.txt		 11B	 1
rand.txt 30.49MiB 1
^PUT.*=> ais://$BUCKET_1/object_1.txt$

^PUT.*=> ais://$BUCKET_1/object_3.txt$

NAME			 SIZE	 VERSION
cksum		 11B	 4
object_1.txt		 2B  2
object_2.txt		 11B     1
object_3.txt         11B     1
rand.txt 30.49MiB 1
"ais://$BUCKET_2" created
^GET.*from ais://$BUCKET_2.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
25
^GET.*from ais://$BUCKET_2.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
25
"ais://$BUCKET_3" created
^GET.*from ais://$BUCKET_3.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
9
^GET.*from ais://$BUCKET_3.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
9
"ais://$BUCKET_4" created
^GET.*from ais://$BUCKET_4.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
25
^GET.*from ais://$BUCKET_4.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
25
"ais://$BUCKET_5" created
^GET.*from ais://$BUCKET_5.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
25
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
^GET.*$
25
